client/consensus/grandpa/authorities.go

```diff
@@ -33,11 +33,11 @@ type delayedKinds[N constraints.Unsigned] interface {
 
 // delayKind struct to represent delayedKinds
 type delayKind struct {
-	value interface{}
+	Value interface{}
 }
 
 func setDelayKind[N constraints.Unsigned, T delayedKinds[N]](delayKind *delayKind, val T) {
-	delayKind.value = val
+	delayKind.Value = val
 }
 
 func newDelayKind[N constraints.Unsigned, T delayedKinds[N]](val T) delayKind {
```

Comment:
Ok, I think I left this functions passes in some previous review, but now I'm wondering why we need those generics if the `delayKind` type holds an interface.

I would suggest to change to

```go

type delayKind[N constraints.Unsigned, T delayedKinds[N]] struct {
	Value T
}

func newDelayKind[N constraints.Unsigned, T delayedKinds[N]](val T) delayKind[N, T] {
	return delayKind[N, T]{
		Value: val,
	}
}
```

Then you can remove `setDelayKind`

Reply:
And then when you want to to switch over a type you can use:

```go
switch any(pending.DelayKind.Value).(type) {
	case Finalized:
		return authSet.addStandardChange(pending, isDescendentOf)
	case Best[N]:
		return authSet.addForcedChange(pending, isDescendentOf)
	default:
		panic("DelayKind is invalid type")
}
```

or even

```go
switch v := any(change.DelayKind.Value).(type) {
...
```

Reply:
So while I like this suggestion, adding another generic constraint to the PendingChange would be a divergence from how substrate does it, so I think I would rather have the constraints match substrates in this case and leave as is.

However, if you still think it would be better to make the change I can.

Reply:
I think this conversation makes more sense in this PR #3283.  The code that introduces this type is there.

Reply:
I'm not sure what the suggestion is? I thought we do have these type of constructors after you rebased.

Reply:
This has been addressed in #3283